### ISSUE TITLE
Requested feature to support "Most Improved" Awards

### ISSUE THREAD

**Issue by millsjdm** [barberscore/barberscore-api#250]
While we won't (for now) do any automatic calculations as to which group wins these awards - 
For any contest session that includes a "Most Improved" Award and for those groups that have indicated they wish to be eligible for that award -  Provide a field in Session Manager next to each competitor entry that is only editable by the DRCJ. Here, the DRCJ would enter the score for that group from last year, or an aggregate score from the last 3 years, or whatever their district policy is. If the group has no "previous" score to assign, he simply removes them from that Award, and the box disappears. 
We need this to provide a clean mechanism to transfer those "compare to this" scores over to the CAs when they use the legacy software to calculate the most improved winner. Later on - perhaps we can add that "calculating" functionality into Scoring Manager.

**Comment by dbinetti** [comment]
This is now officially on you @millsjdm .  We need each "improved" award categorized into either 'raw' or 'standard', and then I need to know how they are calculated.  We have the data, so now it's just a question of getting it done.

**Comment by millsjdm** [comment]
[Most-Improved Awards.xlsx](https://github.com/barberscore/barberscore-api/files/3066018/Most-Improved.Awards.xlsx)

Both raw and SSF are calculated by having the DRCJ provides a 'base' score number to the CA that represents what will be used to measure against. In some cases, it is their previous score from a year or two ago. In other cases, it is an average of previous contests. In all cases, it is always a 'manual' entry that will need to be input into their appearance/record at some point by the CA.

Calculation for Raw:  The group with the greatest "raw" improvement in score is the winner. If nobody improves in score, there is no winner.

Calculation for SSF (Society Standard Formula): I do not know... but Chris does. I will provide that in a followup comment.

**Comment by dbinetti** [comment]
Thank you.  Since there are no IDs  is there a pattern?  Otherwise @millsjdm you'll need to do this manually for each.  If there's a pattern I can update it

**Comment by millsjdm** [comment]
[Most-Improved Awards v2.xlsx](https://github.com/barberscore/barberscore-api/files/3070510/Most-Improved.Awards.v2.xlsx)
1) Added unique Barberscore Ids for all 63 awards
2) Added a tab in the spreadsheet for a detailed definition of both algorithms

Let me know if you need more info. Thanks!

**Comment by dbinetti** [comment]
What happens with ties using the Raw method?

**Comment by millsjdm** [comment]
Same as ranking ties - higher SNG score.

**Comment by dbinetti** [comment]
I have put in as the "base" score the most recent average but that's an assumption.  Someone will need to review this (only  available on back-end until I can test).  

When those are reviewed then I can test against the already finished contests and if valid I can open on front end and put into production


Ref in ceee44be

**Comment by millsjdm** [comment]
I'll get Chris to help me review against the legacy caluculations. Thanks.

**Comment by dbinetti** [comment]
sure, but what I'm saying is "review the 'base' field for all Entries of
conventions that have already completed this year". You said the DRCJ
provides, right?  So then it goes on the Entry.

**Comment by millsjdm** [comment]
Got it... but I don't have those base scores. They were provided to each CA (so they could setup legacy) just prior to each contest. I should be able to gather then from Chis.

**Comment by dbinetti** [comment]
We need to make this a higher priority so we can test it out in a production environment before next year.  Ping @millsjdm @CLBuechler

**Comment by CLBuechler** [comment]
What two contests would you like base data on for testing?

**Comment by dbinetti** [comment]
Is there any particular reason we can't include this for all remaining
contests?

**Comment by CLBuechler** [comment]
Thought you wanted to test some.

What is interface to enter base scores? And No you can’t just pull from prior contest.

Chris

**Comment by CLBuechler** [comment]
I looked at the Base figures in NED Western. Presumably these are taken from the most recent contest appearance - without regard to when or how long ago.  I understand this from programmatic perspective.  But until/unless we get complete buy-in from districts, I do not think it should be imposed on them.

— Chris

**Comment by millsjdm** [comment]
Agreed. We need to continue to be able to support each district's policy.
Therefore, it will need to be a manual entry provided by the DRCJ. Perhaps
we could implement (as a default base score) the group's previous score...
then the DRCJ can change (or leave as-is) as they wish.

David Mills - (813) 786-4381
[sent from my Galaxy Note 9]

**Comment by dbinetti** [comment]
I did it to help with data entry; do you want me to wipe them all?  Please
confirm, I'm trying to ease work for you.

I also put the 'base' field on the Entry inline from the Session -- this
will allow you to enter them in all at once from one page, rather than
going page to page (which is a pain)

**Comment by dbinetti** [comment]
As to which ones, I guess it makes sense for one contest that has
already happened for each award type (raw and standard).  But ideally
I'd like it for the whole season just because it's difficult to
foresee where things will break....  I'd hate for the first time we do
this to be when BS is primary.
